### ISSUE TITLE
Add guide: data extraction example

### ISSUE THREAD

**Issue by brainlid** [brainlid/langchain#97]
Create a documentation guide for structured data extraction. Why the choices were made and example usages.

Include references to docs like this: https://docs.anthropic.com/claude/docs/tool-use#best-practices-for-tool-definitions with quotes to "Provide extremely detailed descriptions." and how descriptions are more valuable than examples.

**Comment by cpursley** [comment]
> US state names to abbreviations

This is actually a great example. One of the things I've built (with ExInstructor) was a US address normalizer. Took a while to whip it into something useful/reliable.

**Comment by brainlid** [comment]
Yeah, I actually thought of doing a demo of that. However, I don’t have a
particular need for it myself. If you don’t mind sharing it, we could
create it as a demo, guide, or livebook example
